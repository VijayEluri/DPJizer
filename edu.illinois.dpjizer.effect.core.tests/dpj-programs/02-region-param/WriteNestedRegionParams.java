/**
 * This file is licensed under the University of Illinois/NCSA Open Source License. See LICENSE.TXT for details.
 */
package regionparam.writenestedregionparams;

/**
 * 
 * @author Mohsen Vakilian
 * 
 */
class C2<region R2> {
	public int x in R2 = 5;
	
	C2() {}
}

class C1<region R1> {
	region r1;
	C2<r1> c2;

	C1() {}
	
	void m() {
		c2 = new C2<r1>();
		c2.x = 15;
	}
}

