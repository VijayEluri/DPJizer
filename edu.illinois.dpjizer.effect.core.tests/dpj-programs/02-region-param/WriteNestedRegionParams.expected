regionparam.writenestedregionparams.C1.C1() invokes java.lang.Object.Object()
regionparam.writenestedregionparams.C1.m() writes Root, Root : regionparam.writenestedregionparams.C1.r1
regionparam.writenestedregionparams.C2.C2() invokes java.lang.Object.Object()
